### PR TITLE
US621093: Create Tomcat Default Logging image for Java 11

### PR DESCRIPTION
**Pull request by tbroadbent22**
https://internal.almoctane.com/ui/entity-navigation?p=131002/6001&entityType=work_item&id=621093

### CODE REVIEW

**Review by frankmccarry — CHANGES_REQUESTED** [review]
This file is still present, is it required;
\opensuse-tomcat-jre11-image\src\main\docker\startup.d\setup-tomcat-ssl-cert.sh
Is the jar log4j-over-slf4j required in the build, it was not present in the jre8 version?

**Comment by tbroadbent22** [comment]
> This file is still present, is it required; \opensuse-tomcat-jre11-image\src\main\docker\startup.d\setup-tomcat-ssl-cert.sh Is the jar log4j-over-slf4j required in the build, it was not present in the jre8 version?

The first issue there has now been addressed and pending a dev build completing.
On the second issue, I did notice that in the [JRE8 pom](https://github.com/CAFapi/opensuse-tomcat-image/blob/develop/pom.xml) log4j-over-slf4j is not present however it is present in the [JRE11 pom](https://github.com/CAFapi/opensuse-tomcat-jre11-image/blob/master/pom.xml#L151) prior to any of my changes, I removed it and built locally on which the project failed, so it looks like it is required although has been removed from the JULI image

File: opensuse-tomcat-jre11-juli-image/pom.xml [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+                </executions>
+                <configuration>
+                    <images>
+                        <image>
+                            <name>${dockerCafImagePrefix}opensuse-tomcat-jre11-juli${dockerProjectVersion}</name>
+                            <build>
+                                <dockerFileDir>.</dockerFileDir>
+                                <assembly>
+                                    <basedir>/</basedir>
+                                </assembly>
```
Comment (frankmccarry) [inline opensuse-tomcat-jre11-juli-image/pom.xml:65]: Did you have a purpose for this entry, it creates a warning in the build logs.

**Review by frankmccarry — CHANGES_REQUESTED** [review]

File: opensuse-tomcat-jre11-juli-image/pom.xml (hunk 1 above)
Comment (tbroadbent22) [reply opensuse-tomcat-jre11-juli-image/pom.xml:65]: I was following what was done for [JRE8 here](https://github.com/CAFapi/opensuse-tomcat-image/blob/develop/opensuse-tomcat-juli-image/pom.xml#L99), as explained earlier, there are no tests as part of this image currently, there are however in JRE8

Comment (tbroadbent22) [reply opensuse-tomcat-jre11-juli-image/pom.xml:65]: What was the warning log so can test/look out for it?

Comment (frankmccarry) [reply opensuse-tomcat-jre11-juli-image/pom.xml:65]: > I was following what was done for [JRE8 here](https://github.com/CAFapi/opensuse-tomcat-image/blob/develop/opensuse-tomcat-juli-image/pom.xml#L99), as explained earlier, there are no tests as part of this image currently, there are however in JRE8

Hi Tony, this is not related to the tests that were added in JRE8. Take a look at what this assembly block is doing in the pom and then take a look for the corresponding warning in the log files. Once you define / assemble files in the pom you need to use the assembled file/artifact in the docker file.

**Review by frankmccarry — APPROVED** [review]
good thanks

**Pull request merged**